ARKitFullExample/View Controllers/SKViewController.swift

```diff
@@ -0,0 +1,218 @@
+        SKSceneView.showsNodeCount = true
+        
+        // Load the SKScene from 'Scene.sks'
+        if let scene = SKScene(fileNamed: "Scene") {
+            SKSceneView.presentScene(scene)
+        }
+        
+        /*----👇---- ARVideoKit Configuration ----👇----*/
+
+        // Initialize ARVideoKit recorder
+        recorder = RecordAR(ARSpriteKit: SKSceneView)
+        // Set the renderer's delegate to retrieve the rendered buffers
+        recorder?.renderAR = self
+        // Enable rendering pre-recording (to push buffers to TwilioVideo)
+        recorder?.onlyRenderWhileRecording = false
+        // Set frames per second rate
+        recorder?.fps = .fps30
+        
+        
+        /*----👇---- TwilioVideo Configuration ----👇----*/
+
+        self.videoTrack = TVILocalVideoTrack.init(capturer: self)
+        self.audioTrack = TVILocalAudioTrack.init()
+        
+        let options = TVIConnectOptions.init(token: accessToken, block: {(builder: TVIConnectOptionsBuilder) -> Void in
+            if let videoTrack = self.videoTrack {
+                builder.videoTracks = [videoTrack]
+            }
+            if let audioTrack = self.audioTrack {
+                builder.audioTracks = [audioTrack]
+            }
+            builder.roomName = "arkit"
+        })
+        
+        self.room = TwilioVideo.connect(with: options, delegate: self)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        // Create a session configuration
+        let configuration = ARWorldTrackingConfiguration()
+        
+        // Run the view's session
+        SKSceneView.session.run(configuration)
+        
+        // Prepare the recorder with sessions configuration
+        recorder?.prepare(configuration)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        // Pause the view's session
+        SKSceneView.session.pause()
+        
+        recorder?.onlyRenderWhileRecording = true
+        recorder?.prepare(ARWorldTrackingConfiguration())
+        
+        // Switch off the orientation lock for UIViewControllers with AR Scenes
+        recorder?.rest()
+
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    @IBAction func goBack(_ sender: UIButton) {
+        self.dismiss(animated: true, completion: nil)
+    }
+}
+
+//MARK: - ARVideoKit Renderer Method
+extension SKViewController: RenderARDelegate {
+    func frame(didRender buffer: CVPixelBuffer, with time: CMTime, using rawBuffer: CVPixelBuffer) {
+        if isStreamingVideo {
+            self.frame = TVIVideoFrame(timestamp: Int64(CMTimeGetSeconds(time)),
+                                       buffer: buffer,
+                                       orientation: TVIVideoOrientation.up)
```

Comment:
`TVIVideoFrame` takes timestamp in nano-seconds - [example usage](https://github.com/twilio/video-quickstart-swift/blob/master/ARKitExample/ViewController.swift#L91-L93)